src/distributions/weighted.rs

```diff
@@ -131,40 +133,52 @@ impl<X> Distribution<usize> for WeightedIndex<X> where
 }
 
 #[allow(missing_docs)] // todo: add docs
-#[derive(Debug, Clone)]
-pub struct AliasMethodWeightedIndex {
+#[allow(missing_debug_implementations)] // todo: why does `#[derive(Debug)]` not work?
```

Comment:
You probably would have to require `W` to implement `Debug`.

Reply:
I tried that but it didn't work.

Reply:
It would probably work if you force `trait UniformSampler: Debug + Sized`, except probably other stuff would break... try:
```
where W: Debug, super::Uniform<W>: Debug
```

Reply:
What were you trying to do though? Use the `derive(Debug)` macro? That [cannot currently add the correct bounds](https://github.com/rust-lang/rust/issues/26925). I think to implement `Debug` correctly here you would need to do so manually (with the above `where` clause).

Reply:
Ah, I wasn't aware of https://github.com/rust-lang/rust/issues/26925. I have implemented `Debug` and `Clone` myself.